pkg/infra/initiator.go

```diff
@@ -0,0 +1,32 @@
+package infra
+
+import (
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+)
+
+func StartCreateProposal(num int, size int, r float64, config Config, crypto *Crypto, raw chan *Elements, errorCh chan error, logger *log.Logger) {
+	limit := rate.Inf
+	if r > 0 {
+		logger.Infof("Will use rate %f as send rate\n", r)
+		limit = rate.Limit(r)
+	}
+	limiter := rate.NewLimiter(limit, size)
+	for i := 0; i < num; i++ {
+		prop, err := CreateProposal(
+			crypto,
+			config.Channel,
+			config.Chaincode,
+			config.Version,
+			config.Args...,
+		)
+		if err != nil {
+			errorCh <- errors.Wrapf(err, "error creating proposal")
+			return
+		}
+		if limiter.Allow() {
```

Comment:
I don't think `Allow` is correct here. We should be using `Wait`, no?

Reply:
https://pkg.go.dev/golang.org/x/time@v0.0.0-20201208040808-7e3f01d25324/rate?utm_source=gopls#Limiter.ReserveN

Either `Allow` or `Wait` invokes `Reserve`.
Allow just return bool, so it will running is 
```
if (allow) {
act()
} else {
continue
}
```
Well, at performance point of view, will update code to 
```
r := lim.ReserveN(time.Now(), 1)
if !r.OK() {
  // Not allowed to act! Did you remember to set lim.burst to be > 0 ?
  return
}
time.Sleep(r.Delay())
Act()
```

Reply:
Yep, change to `Wait` as fix.

Reply:
change to `Reserve()`

```
		r := limiter.Reserve()
		if !r.OK() {
			return
		}
		time.Sleep(r.Delay())
```